src/Stratis.Bitcoin.IntegrationTests/Utilities/Extensions/CoreNodeExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Consensus;
+using Stratis.Bitcoin.Features.Consensus.Interfaces;
+using Stratis.Bitcoin.IntegrationTests.EnvironmentMockUpHelpers;
+
+namespace Stratis.Bitcoin.IntegrationTests.Utilities.Extensions
+{
+    public static class CoreNodeExtensions
+    {
+        public static Money CalculateProofOfWorkReward(this CoreNode node, int blockCount)
+        {
+            var consensusValidator = node.FullNode.NodeService<IPowConsensusValidator>() as PowConsensusValidator;
+
+            var groupedRewardsBySubsidyHalving = Enumerable.Range(0, blockCount)
```

Comment:
here, you did 0-blockcount. This implies to me that this will only work when using it to validate a reward when you started mining at from 0

Reply:
This is just building up a sequence of numbers to work out each reward per group - I tested if zero blocks are mined and there is no side effect on this function.